wordpress-mobile/gutenberg-mobile
# Add video block ( first iteration )

To Fix: https://github.com/wordpress-mobile/gutenberg-mobile/issues/855

There are also some more TODO items in this issue: https://github.com/wordpress-mobile/gutenberg-mobile/issues/687#issuecomment-482011555 which are not handled in this PR

This PR introduces video block.

Gutenberg PR: https://github.com/WordPress/gutenberg/pull/14912

This is a breaking change for parent apps:

WPiOS PR: https://github.com/wordpress-mobile/WordPress-iOS/pull/11454
WPAndroid PR: https://github.com/wordpress-mobile/WordPress-Android/pull/9644

iOS Example app:

![video-block-ios2](https://user-images.githubusercontent.com/5032900/56147936-ece82e00-5fb1-11e9-81db-c1cc4676b25d.gif)

Android Example app:
(Media option picker is not working but a sample video block is added to content)

![video-block-android](https://user-images.githubusercontent.com/5032900/55906027-fea98a00-5bdb-11e9-8f6c-096afa0d21b9.gif)


## Types of changes

- Integrated react-native-video library to both iOS and Android example apps
- Updated bridge to pass media `filter` such as 'video', 'image'
- Added a sample video block to example app content

**To Test**

Test with both iOS and Android Example apps:

**iOS Example App Tests**

Note: Video playback isn't very smooth in iOS simulator but there's no problem on real devices as far as I can tell.

**Test 1** - Add Video block

- Open the media picker and add video block
<br>

**Test 2** - Insert video from device (successful) (currently available on iOS example app only)

- Tap on the Video block placeholder
- Choose option 'Choose from device'
- Verify that you can choose among videos only(in order to add videos to the ios simulator just drag a video from your computer and drop to the simulator)
- Choose a video
- Verify that spinner appears to show the progress:
<img width="334" alt="Screen Shot 2019-04-15 at 18 27 44" src="https://user-images.githubusercontent.com/5032900/56145064-3d5c8d00-5fac-11e9-8383-0cc2563b8454.png">
- At the end of the progress you should be able to tap and play the video
<br>

**Test 3** - Insert video from device (failing) (currently available on iOS example app only)

- Select the video block
- Tap edit button in the toolbar
- Choose option 'Choose from device', then choose a video
- During the spinner upload continues, longpress somewhere in the screen(longpress will trigger mocking of upload failure)
- Verify you see the retry message:
<img width="337" alt="Screen Shot 2019-04-15 at 18 33 49" src="https://user-images.githubusercontent.com/5032900/56145477-09359c00-5fad-11e9-8e50-167da89d1d5c.png">
- Tap on the gray area and verify that you see the retry option:
<img width="347" alt="Screen Shot 2019-04-15 at 18 34 17" src="https://user-images.githubusercontent.com/5032900/56145596-40a44880-5fad-11e9-90ae-318071631224.png">
- Tap on Retry and see that upload starts again 
<br>

**Test 4** - Insert video from device (cancel) (currently available on iOS example app only)
- Select the video block
- Tap edit button in the toolbar
- Choose option 'Choose from device', then choose a video
- During the spinner upload tap on the video block 
- Verify that Cancel option appears:
<img width="344" alt="Screen Shot 2019-04-15 at 19 04 09" src="https://user-images.githubusercontent.com/5032900/56147620-4c920980-5fb1-11e9-952d-83d1afde9c00.png">
- Choose Cancel and see that media is cleared and you see the placeholder again.
<br>

**Test 5** - Insert video from WP media library (currently available on iOS example app only)

- Select the video block
- Tap edit button in the toolbar
- Choose option 'WordPress media library'
(Since this is the example app we are only updating the block with a video link)
- Verify that video block is updated with another video and you can play the new video
<br>

**Test 6**

Verify that above tests are working for Image block also
